src/test/java/org/jitsi/meet/test/base/Participant.java

```diff
@@ -115,30 +99,26 @@ public Participant(
      */
     public void joinConference(String roomName)
     {
-        this.joinConference(roomName, null, null, null);
+        this.joinConference(roomName, new JitsiMeetUrl());
     }
 
     /**
      * Joins a conference.
      *
      * @param roomName the room name to join.
-     * @param roomParameter the {@link JitsiMeetUrl#roomParameters} part of
-     * the Jitsi Meet URL.
-     * @param config the {@link JitsiMeetUrl#hashConfigPart} which will be
-     * appended at the end of the default config part (if any) specified in the
      * {@link Participant}'s constructor.
      */
     public void joinConference(
-        String roomName, String roomParameter, String config,
-        Consumer<JitsiMeetUrl> customJoinImpl)
+        String roomName,
+        JitsiMeetUrl meetURL)
     {
-        JitsiMeetUrl conferenceUrl = (JitsiMeetUrl) this.meetURL.clone();
-
-        conferenceUrl.setRoomName(roomName);
-        conferenceUrl.setRoomParameters(roomParameter);
-        conferenceUrl.appendConfig(config);
+        meetURL = Objects.requireNonNull(meetURL, "meetURL");
+        if (meetURL.getRoomName() == null)
+        {
+            meetURL.setRoomName(roomName);
+        }
 
```

Comment:
This check for not overriding the room name is needed for the iframetest. Otherwise it overrides the roomname and breakes the iframe link and we see access denied.

Reply:
can we get rid of the roomName argument in joinConference ?

Reply:
Yep, I was thinking the same, but didn't have time to try it. I suppose we can get rid of it :)